src/ClassValidatorMiddleware.spec.ts

```diff
@@ -36,6 +36,73 @@ describe("ClassValidatorMiddleware", () => {
           lastName: "Doe",
         });
       });
+
+      it("sets the body to the transformed and validated value with bodyClassType param", async () => {
+        const handler = {
+          callback: jest.fn(),
+          context: {} as Context,
+          error: {} as Error,
+          event: { body },
+          internal: {},
+          response: {},
+        };
+        await classValidatorMiddleware({
+          bodyClassType: NameBody,
+        }).before(handler);
+        expect(handler.event.body).toEqual({
+          firstName: "John",
+          lastName: "Doe",
+        });
+      });
+    });
+
+    describe("with invalid input", () => {
+      const body = JSON.stringify({
+        firstName: "John",
+      });
+
+      it("throws an error with statusCode 400", async () => {
+        const handler = {
+          context: {} as Context,
+          error: {} as Error,
+          event: { body },
+          internal: jest.fn(),
+          response: {},
+        };
+        await expect(
+          classValidatorMiddleware({
+            classType: NameBody,
+          }).before(handler),
+        ).rejects.toMatchObject({
+          statusCode: 400,
+        });
+      });
+    });
+  });
+
+  describe("before hook with query params", () => {
+    describe("with valid input", () => {
+      const queryStringParameters = JSON.stringify({
+        firstName: "John",
+        lastName: "Doe",
+      });
```

Comment:
double-checking: is the input we get in a lambda here a string, or is it the actual object?

Reply:
It will be a json string only.